plugin/secrets_service_account_key.go

```diff
@@ -232,3 +403,84 @@ would generate service account keys for the "deploy" role set.
 On the backend, each roleset is associated with a service account under
 which secrets/keys are created.
 `
+
+type serviceAccountKeyCacheCollection struct {
+	Items map[string]*serviceAccountKeyCacheItem
+}
+
+func newServiceAccountKeyCacheCollection(item *serviceAccountKeyCacheItem, itemKey string) *serviceAccountKeyCacheCollection {
+	cacheCollection := new(serviceAccountKeyCacheCollection)
+	cacheCollection.Items = make(map[string]*serviceAccountKeyCacheItem)
+
+	if item != nil && itemKey != "" {
+		cacheCollection.Items[itemKey] = item
+	}
+
+	return cacheCollection
+}
+
+func (c *serviceAccountKeyCacheCollection) getFirstValidItem(rsBindingHash string) (string, *serviceAccountKeyCacheItem) {
+	keys := make([]string, 0, len(c.Items))
+	for k := range c.Items {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
```

Comment:
Shouldn't we sort this in descending manner?

Also, you might want to give some example on the values of `keys` so we know exactly what to expect after we sort it

Reply:
Yeah, I just realized that. I'll update it in the next commit.